service/wave/wave.go

```diff
@@ -0,0 +1,334 @@
+package wave
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/client"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/util/uritemplates"
+)
+
+type Cluster struct {
+	ID                *string      `json:"id,omitempty"`
+	ClusterIdentifier *string      `json:"clusterIdentifier,omitempty"`
+	Environment       *Environment `json:"environment,omitempty"`
+	Config            *Config      `json:"config,omitempty"`
+	State             *string      `json:"state,omitempty"`
+
+	// Read-only fields.
+	CreatedAt *time.Time `json:"createdAt,omitempty"`
+	UpdatedAt *time.Time `json:"updatedAt,omitempty"`
+}
+
+type Environment struct {
+	OperatorVersion         *string `json:"operatorVersion,omitempty"`
+	CertManagerDeployed     *bool   `json:"certManagerDeployed,omitempty"`
+	K8sClusterProvisioned   *bool   `json:"k8sClusterProvisioned,omitempty"`
+	OceanClusterProvisioned *bool   `json:"oceanClusterProvisioned,omitempty"`
+	EnvironmentNamespace    *string `json:"environmentNamespace,omitempty"`
+	OceanClusterId          *string `json:"oceanClusterId,omitempty"`
+}
+
+type Config struct {
+	Components []*Component `json:"components,omitempty"`
+}
+
+type Component struct {
+	Uid             *string           `json:"uid,omitempty"`
```

Comment:
```suggestion
	UID             *string           `json:"uid,omitempty"`
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)